src/core/Form/Select/BaseSelect/SelectItemList/SelectItemList.tsx

```diff
@@ -23,9 +22,7 @@ export interface SelectItemListProps {
   /** SelectItemList container div class name for custom styling. */
   className?: string;
   /** List items */
-  children:
-    | Array<ReactElement<SelectItemProps>>
-    | ReactElement<SelectItemProps>;
+  children: ReactElement | Array<ReactElement>;
```

Comment:
I simplified this a little bit since the different types of elements this can take as children is already quite long with the addition of <SelectItemAddition>. Also it was quite difficult to make the types work with conditional rendering logic. 

I say this should be fine since this interface is only for our internal use.